### ISSUE TITLE
Can't save new empty files.

### ISSUE THREAD

**Issue by ArthurFabian** [notepad-plus-plus/notepad-plus-plus#4505]
### Description of the Issue

Can't save a newly created empty file.

### Steps to Reproduce the Issue

1. Ctrl + N 
2. Ctrl + S

### Expected Behavior

"Save as" popup window

### Actual Behavior

Nothing

**Comment by sasumner** [comment]
In this scenario there is nothing to save.  Note that after step 1 the disk icon on the "new #" editor tab is blue (which indicates there is nothing to save).  If you wish for your steps to give you the popup, in between step 1 and 2 type a character into the editor window and then press Backspace.  After that the icon will be red and initiating a save will cause the popup to happen.  Please close this issue.

**Comment by ArthurFabian** [comment]
Why does user have to type character and hit back space to save? Seems counter intuitive.

**Comment by sasumner** [comment]
Because there is nothing to save before making some kind of change.  :)

**Comment by fml2** [comment]
I'd rather agree with @ArthurFabian. The save button should do nothing if the window contents is in sync with the disc. Note that it's not the same as "nothing has been changed in the window": Just after creating a new file, there are no changes in the window, but this state is still not present on the disc, hence no sync, hence the Save button should bring up the Save dialog.

I undestand that it's not a proof, but MS Word and Wordpad work exactly like this.

**Comment by ArthurFabian** [comment]
Any text editor I've used works like this. I haven't encountered any other inconveniences so far with notepad++, which is great! Admittedly this one is pretty tame, but it still interrupts cognitive flow. I'll try to fix it myself if someone can help me figure it out. I found the code for new empty document here: 

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/6c9a5a2afe02c7ed578a26855f1daac76f3d9be4/PowerEditor/src/ScitillaComponent/Buffer.cpp#L1186

I'm just not sure how to make the document savable. Any idea?

**Comment by fml2** [comment]
`setDirty(true)`?

**Comment by webmanoffesto** [comment]
**File / New, File / Save** As works
**Ctrl-N, Ctrl-S** doesn't work

**Comment by sasumner** [comment]
OMG!

**Comment by hugobuddel** [comment]
I often like to give the file a name before typing any contents. Every editor can do this.

So I always do Ctrl-N, space, Ctrl-S, save the file, backspace, Ctrl-S. This is quite annoying. I could do the backspace before saving to save some time, but the instinct to save is apparently quite strong.

Note that "There is nothing to save" is an invalid argument, because space+backspace also results in a file where there is nothing to save. (Uh oh, hopefully saving such a file won't be disabled either..)

But, this discussion has given me the insight I need: Ctrl-N, Ctrl-Alt-S, because save-as does work. Still a bit annoying, but significantly less so.

**Comment by sasumner** [comment]
I think it works on a "dirty" mechanism.  If it isn't dirty, it can't be saved.  And maybe it eases annoyance if you create a new file and then change your mind and want to just get rid of it quickly (without prompting).

Regardless, all could be coded around and your suggestion is a good one.

>Ctrl-N, Ctrl-Alt-S

Nice workaround. :)

**Comment by hugobuddel** [comment]
> I think it works on a "dirty" mechanism. If it isn't dirty, it can't be saved. And maybe it eases annoyance if you create a new file and then change your mind and want to just get rid of it quickly (without prompting).

Interesting though: a file is dirty if you create it, type a space, and backspace. But it will close without prompt with Ctrl-W. So new, empty, and dirty files can already be closed quickly. Automatically making a new file dirty seems to have no bad side effects.

**Comment by donho** [comment]
The current behaviour is intentional and will be kept.

**Comment by sasumner** [comment]
Appears the workaround has now ratcheted up several notches in importance! :)

**Comment by hugobuddel** [comment]
It would be nice to have an explanation behind the intent. Because we failed in guessing the intent:

- The behavior is different from any other application (and thus different from user expectations).
- None of us could think of a scenario where the current behavior would be desirable.
- Every hypothetical situation we could come up with, would also work fine with dirty new files.

But whatever, the workaround works and we're not entitled to anything.
